pandas/tests/sparse/test_array.py

```diff
@@ -454,6 +454,17 @@ def test_values_asarray(self):
         assert_almost_equal(self.arr.to_dense(), self.arr_data)
         assert_almost_equal(self.arr.sp_values, np.asarray(self.arr))
 
+    def test_shape(self):
+        # GH 21126
```

Comment:
can you add cases that have a fill_value, can you parameterize the tests. also add a test with an object array as well (e.g. steal cases from the constructor tests)